terkin/datalogger.py

```diff
@@ -273,7 +275,17 @@ def get_sleep_time(self):
         self.settings.setdefault('main.interval.maintenance', 5.0)
 
         # Compute interval.
-        interval = self.settings.get('main.interval.field')
+        # pycom.nvs_get should return None in case of unset key. Instead it throws an error
+        try:
+            _deepsleep = pycom.nvs_get('deepsleep')
+        except Exception as ex:
+            _deepsleep = None
+
+        if _deepsleep is None:
+            interval = self.settings.get('main.interval.field')
+        else:
+            interval = pycom.nvs_get('deepsleep') * 60
+            log.info('DEEP SLEEP interval set to %s minute(s) by LoRa downlink message', int(interval / 60))
```

Comment:
Can I humbly ask you streamline this a bit like:
```
# First, try to acquire deep sleep interval from NVRAM.
# This gets used when set from a LoRaWAN downlink message.
try:
    interval_minutes = pycom.nvs_get('deepsleep')
    log.info('Deepsleep interval set to %s minute(s) by LoRaWAN downlink message', interval_minutes)
    interval = interval_minutes * 60

# Otherwise, fall back to original configuration setting.
except Exception as ex:
    interval = self.settings.get('main.interval.field')
```

Reply:
resolved with https://github.com/hiveeyes/terkin-datalogger/pull/49/commits/b213885180cb3bafb7614387dbfc04f4fe6372eb